File: 스티커.cpp
#include<iostream>
#include<vector>
using namespace std;
int n, t;
long long int dp[2][100000];
int main() {
	cin >> t;
	while (t--) {
		cin >> n;
		vector<vector<int>>v(2, vector<int>(n));
		for (int i = 0; i < 2 * n; i++) {
			if (i < n) cin >> v[0][i];
			else cin >> v[1][i - n];
		}

		dp[0][0] = v[0][0];
		dp[1][0] = v[1][0];
		if (n > 1) {
			dp[0][1] = dp[1][0] + v[0][1];
			dp[1][1] = dp[0][0] + v[1][1];

			for (int i = 2; i < n; i++) {
				dp[1][i] = max(dp[0][i - 1], dp[0][i - 2]) + v[1][i];
				dp[0][i] = max(dp[1][i - 1], dp[1][i - 2]) + v[0][i];
			}
		}
		cout << max(dp[0][n - 1], dp[1][n - 1]);
	}
	return 0;
}

File: 스티커.py
import sys
input = sys.stdin.readline
t = int(input())
for _ in range(t):
    n = int(input())
    dp = [[0] * n for _ in range(2)]
    arr = []
    
    ret = 0
    for _ in range(2):
        arr.append(list(map(int,input().split())))
    for i in range(n):
        if i == 0:
            dp[0][i] = arr[0][i]
            dp[1][i] = arr[1][i]
        elif i == 1:
            dp[0][i] = dp[1][i-1] + arr[0][i]
            dp[1][i] = dp[0][i-1] + arr[1][i]
        else:
            dp[0][i] = max(dp[0][i-2] + arr[0][i], dp[1][i-2] + arr[0][i], dp[1][i-1] + arr[0][i])
            dp[1][i] = max(dp[1][i-2] + arr[1][i], dp[0][i-2] + arr[1][i], dp[0][i-1] + arr[1][i])
    ret = max(max(dp[0]),max(dp[1]))
    print(ret)